### DISCUSSION TITLE
Linuxbrew: conan@1 causes Error: No available formula with the name "cmake".

### DISCUSSION THREAD

**Discussion by smessmer** [Homebrew/discussions#4580] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
[ as logged by `brew test-bot --only-setup` in the CI job ]

==> brew config
HOMEBREW_VERSION: 4.0.23-12-ge986264
ORIGIN: https://github.com/Homebrew/brew
HEAD: e986264a3e5f4cb14cbc98efbc3c0e09aa363b9f
Last commit: 2 days ago
Core tap: N/A
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_BOOTSNAP: set
HOMEBREW_CACHE: /home/runner/.cache/Homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS: 3650
HOMEBREW_COLOR: set
HOMEBREW_CURL_PATH: /usr/bin/curl
HOMEBREW_DEVELOPER: set
HOMEBREW_FAIL_LOG_LINES: 150
HOMEBREW_GIT_EMAIL: 1589480+BrewTestBot@users.noreply.github.com
HOMEBREW_GIT_NAME: BrewTestBot
HOMEBREW_GIT_PATH: /usr/bin/git
HOMEBREW_LOGS: /home/runner/work/homebrew-tap/homebrew-tap/logs
HOMEBREW_MAKE_JOBS: 2
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_EMOJI: set
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
Homebrew Ruby: 2.6.10 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: dual-core 64-bit skylake
Clang: 14.0.0
Git: 2.40.1 => /usr/bin/git
Curl: 7.81.0 => /usr/bin/curl
Kernel: Linux 5.15.0-1039-azure x86_64 GNU/Linux
OS: Ubuntu 22.04.2 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: 3.0.2
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
[ as logged by `brew test-bot --only-setup` in the CI job ]

==> brew doctor
All steps passed!
```

### Description of issue

I maintain a custom homebrew tap at https://github.com/cryfs/homebrew-tap. To fix [another issue](https://github.com/cryfs/homebrew-tap/issues/14), we had to change the build dependency from "conan" to "conan@1". See https://github.com/cryfs/homebrew-tap/pull/15. The tap repository uses Github Actions with `Homebrew/actions` to test changes.

While the formula without that change [passes CI](https://github.com/cryfs/homebrew-tap/actions/runs/5305175882/jobs/9601966806), the formula with that change [passes CI on macos](https://github.com/cryfs/homebrew-tap/actions/runs/5305357104/jobs/9602267022?pr=15) but [fails on linuxbrew](https://github.com/cryfs/homebrew-tap/actions/runs/5305357104/jobs/9602266984?pr=15) with

```
  /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/bin/bundle clean
  Error: No available formula with the name "cmake".
```

Any idea what is causing this and how I could fix it? Or at least get more information about what's going on? I tried adding a `brew install cmake` command to the beginning of the CI job. That executed fine, installed cmake, but then later still failed with the same error message.

**Comment by ZhongRuoyu** [comment, marked as answer]
This is a known issue that's being tracked at Homebrew/brew#15049. As a workaround, please try adding this to your workflow:

```yaml
env:
  HOMEBREW_NO_INSTALL_FROM_API: 1
```

See the linked issue for more details.

**Comment by smessmer** [comment]
Thanks, that worked :)
